daffodil-test/src/test/resources/org/apache/daffodil/charsets/CustomCharset.tdml

```diff
@@ -0,0 +1,151 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<tdml:testSuite xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:dfdlx="http://www.ogf.org/dfdl/dfdl-1.0/extensions"
+  xmlns:ex="http://example.com"
+  xmlns:tns="urn:org.apache.daffodil.charsets.CustomCharset" defaultRoundTrip="none">
+
+  <tdml:parserTestCase name="parse_charsets" root="s1" model="org/apache/daffodil/charsets/CustomCharset.dfdl.xsd">
+
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <s1>
+          <e1>À1234567</e1>
+          <e2>01234567</e2>
+        </s1>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+    <tdml:document>
+      <tdml:documentPart type="byte"><![CDATA[C0 31 32 33 34 35 36 37
+                                            CF CE CD CC CB CA C9 C8]]>
+      </tdml:documentPart>
+    </tdml:document>
+
+  </tdml:parserTestCase>
+
+  <tdml:unparserTestCase name="unparse_charsets" root="s1" model="org/apache/daffodil/charsets/CustomCharset.dfdl.xsd">
+
+    <tdml:document>
+      <tdml:documentPart type="byte"><![CDATA[C0 31 32 33 34 35 36 37
+                                          CF CE CD CC CB CA C9 C8]]>
+      </tdml:documentPart>
+    </tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <tns:s1>
+          <e1>Ą1234567</e1>
+          <e2>01234567</e2>
+        </tns:s1>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+  </tdml:unparserTestCase>
+
+  <tdml:parserTestCase name="parse_charsets2" root="s2" model="org/apache/daffodil/charsets/CustomCharset.dfdl.xsd">
+
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <s2>
+          <e1>À1234567</e1>
+          <e2>À1234567</e2>
```

Comment:
First char inside <e2> should be Ą once you have a real 8859-13 decoder, which I suggest you fix below in comments on the scala code for this charset.